sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1730,8 +1730,6 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
                 .withKeyDeserializerProvider(kafkaRead.getKeyDeserializerProvider())
                 .withValueDeserializerProvider(kafkaRead.getValueDeserializerProvider())
                 .withManualWatermarkEstimator()
-                .withRedistribute()
```

Comment:
this breaks updating the job.

Reply:
I'm not sure that is an argument for fixing this bug because a currently running job on 2.57 is broken with update to 2.58 due to this unconditionally being set.

Reply:
If we believe that Redistribute is a better default, then we could choose to modify this PR to respect the config to allow disabling redistribute.  But since redistribute appears to only work if duplicates are allowed that seems like a dangerous default.

Reply:
shall we consider using https://github.com/apache/beam/blob/9aaf7e41dec8b9d3effdc7cb6c887f8df8e62a64/sdks/java/core/src/main/java/org/apache/beam/sdk/options/StreamingOptions.java#L47 here?

Reply:
If a user launches a pipeline with 2.58 and then upgrades to the fixed 2.59 while passing `--updateCompatibilityVersion=2.58.0` then we have no ability to give them a correct pipeline (changing allow/disallow duplicates flag is not update compatible either). So I don't think that it is worth preserving.

Reply:
We should just put it in the known issues. We might need to mark 2.58.0 as a bad release and issue 2.58.1

Reply:
Added as Known issue to CHANGES.md and blog.

Reply:
Is `withRedistribute` better for the performance if users allow duplicates?

Reply:
Whether or not the performance changes seems orthogonal to respecting the user configuration and defaulting to not introducing duplicates. But it likely depends on the pipeline and what would be fused into reading Kafka if redistribute is better or not.

Reply:
Then shall we at least add the option to allow users to enable redistribute?

Reply:
There is such an option and it is used correctly a few lines below.  The bug is that we are using redistribute/allowduplicates regardless of that configuration here.

```
        if (kafkaRead.isRedistributed()) {
          readTransform = readTransform.withRedistribute();
        }
        if (kafkaRead.isAllowDuplicates()) {
          readTransform = readTransform.withAllowDuplicates();
        }
```

Reply:
I see now. Thanks!